code/modules/projectiles/projectile/experimental.dm

```diff
@@ -0,0 +1,681 @@
+							if(pos_from == SIDE_WEST)
+								pos_to = SIDE_EAST
+							else
+								pos_to = SIDE_WEST
+							pos_from = SIDE_SOUTH
+						if(SIDE_EAST)
+							bump.dir = EAST
+							if(pos_from == SIDE_NORTH)
+								pos_to = SIDE_SOUTH
+							else
+								pos_to = SIDE_NORTH
+							pos_from = SIDE_EAST
+						if(SIDE_WEST)
+							bump.dir = WEST
+							if(pos_from == SIDE_NORTH)
+								pos_to = SIDE_SOUTH
+							else
+								pos_to = SIDE_NORTH
+							pos_from = SIDE_WEST
+
+				else
+					A.bullet_act(src, def_zone)
+					spawn()
+						density = 0
+						invisibility = 101
+						//del(src)
+						loc = null
+						returnToPool(src)
+						OnDeath()
+			else
+				A.bullet_act(src, def_zone)
+				spawn()
+					density = 0
+					invisibility = 101
+					//del(src)
+					loc = null
+					returnToPool(src)
+					OnDeath()
+
+		else if(is_type_in_list(A,ricochet_bump))//beware fuel tanks!
+			A.bullet_act(src)
+			spawn()
+				density = 0
+				invisibility = 101
+				//del(src)
+				loc = null
+				returnToPool(src)
+				OnDeath()
+
+		else if((istype(A,/obj/structure/window) || istype(A,/obj/machinery/door/window)) && (A.loc == src.loc))//all this part is to prevent a bug that causes the shot to go through walls
+																												//if they are next to a one-directional window/windoor.
+			var/stepdir = SOUTH
+			switch(pos_to)
+				if(SIDE_NORTH)
+					stepdir = NORTH
+				if(SIDE_SOUTH)
+					stepdir = SOUTH
+				if(SIDE_EAST)
+					stepdir = EAST
+				if(SIDE_WEST)
+					stepdir = WEST
+			var/turf/T = get_step(src, stepdir)
+			if(T.density)
+				var/obj/structure/ricochet_bump/bump = new(loc)
+				playsound(get_turf(src), 'sound/items/metal_impact.ogg', 50, 1)
+				switch(pos_to)
+					if(SIDE_NORTH)
+						bump.dir = NORTH
+						if(pos_from == SIDE_WEST)
+							pos_to = SIDE_EAST
+						else
+							pos_to = SIDE_WEST
+						pos_from = SIDE_NORTH
+					if(SIDE_SOUTH)
+						bump.dir = SOUTH
+						if(pos_from == SIDE_WEST)
+							pos_to = SIDE_EAST
+						else
+							pos_to = SIDE_WEST
+						pos_from = SIDE_SOUTH
+					if(SIDE_EAST)
+						bump.dir = EAST
+						if(pos_from == SIDE_NORTH)
+							pos_to = SIDE_SOUTH
+						else
+							pos_to = SIDE_NORTH
+						pos_from = SIDE_EAST
+					if(SIDE_WEST)
+						bump.dir = WEST
+						if(pos_from == SIDE_NORTH)
+							pos_to = SIDE_SOUTH
+						else
+							pos_to = SIDE_NORTH
+						pos_from = SIDE_WEST
+
+			else
+				ricochet_jump()
+
+		else
+			ricochet_jump()
+
+/obj/item/projectile/ricochet/process_step()//unlike laser guns the projectile isn't instantaneous, but it still travels twice as fast as kinetic bullets since it moves twices per ticks
+	if(src.loc)
+		if(kill_count < 1)
+			//del(src)
+			OnDeath()
+			loc = null
+			returnToPool(src)
+			return
+		kill_count--
+		for(var/i=1;i<=2;i++)
+			ricochet_movement()
+		sleep(1)
+
+/obj/item/projectile/ricochet/proc/ricochet_movement()//movement through empty space
+	var/stepdir = SOUTH
+	switch(pos_to)
```

Comment:
When you use the real byond defines for north, south, east, west, you can use stepdir = pos_to instead of the switch